### DISCUSSION TITLE
(Automatically) Numbered Headings

### DISCUSSION THREAD

**Discussion by michaelweinold** [microsoft/vscode-jupyter#11315] [New Ideas]
As mentioned by @rchiodo in https://github.com/microsoft/vscode-jupyter/issues/1348#issuecomment-869802932, automatically numbering markdown headings would need to be a separate feature request from the outline view, which was implemented with https://github.com/microsoft/vscode-jupyter/issues/1348.

Expected behaviour: Option to automatically number markdown headings, such that for:

<img width="352" alt="Screenshot 2022-09-07 at 08 57 05" src="https://user-images.githubusercontent.com/23102087/188810116-0d7a9536-b7c0-4b25-838f-51b349ae694f.png">

we would get:

<img width="323" alt="Screenshot 2022-09-07 at 08 57 20" src="https://user-images.githubusercontent.com/23102087/188810120-3be54e0e-0339-4363-847c-b9f240b7ce58.png">

Existing implementation in JupyterLab: [toc2 extension](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/toc2/README.html#)
Existing implementation for pure markdown files: [Markdown All in One: Add/Update section numbers](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one#available-commands)

**Comment by jcsaaver** [comment]
Still hoping for this to be implemented
